hack/golangci.yaml.in

```diff
@@ -185,84 +191,135 @@ linters:
-    rules:
-      - name: exported
-        arguments:
-        - disableStutteringCheck
-  staticcheck:
-    checks:
-      - "all"
-      {{- if .Base }}
-      - "-SA1019"  # TODO(fix) Using a deprecated function, variable, constant or field
-      - "-SA2002"  # TODO(fix) Called testing.T.FailNow or SkipNow in a goroutine, which isn’t allowed
-      - "-SA1006"  # TODO(fix) Using a printf-style function with a dynamic format string and no arguments can just use a print-style func
+  {{- end }}
+
+  settings: # please keep this alphabetized
+    custom:
+      logcheck:
+        # Installed there by hack/verify-golangci-lint.sh.
+        path: ../_output/local/bin/logcheck.so
+        description: structured logging checker
+        original-url: k8s.io/logtools/logcheck
+        settings:
+          config: |
+            {{include "hack/logcheck.conf" | indent 12 | trim}}
+    depguard:
+      rules:
+        main:
+          files:
+            - $all
+            - "!$test"
+            - "!**/test/**"
+          deny:
+            - pkg: "github.com/google/go-cmp/cmp"
+              desc: "cmp is allowed only in test files"
+    forbidigo:
+      analyze-types: true
+      forbid:
+        - pattern: ^managedfields\.ExtractInto$
+          pkg: ^k8s\.io/apimachinery/pkg/util/managedfields$
+          msg: should not be used because managedFields was removed
+        - pattern: \.Extract
+          pkg: ^k8s\.io/client-go/applyconfigurations/
+          msg: should not be used because managedFields was removed
+        - pattern: \.Add$
+          pkg: ^k8s\.io/component-base/featuregate$         
+          msg: should not use Add, use AddVersioned instead 
+      {{- if .Hints}}
+        - pattern: ^gomega\.BeTrue$
+          pkg: ^github.com/onsi/gomega$
+          msg: "it does not produce a good failure message - use BeTrueBecause with an explicit printf-style failure message instead, or plain Go: if ... { ginkgo.Fail(...) }"
+        - pattern: ^gomega\.BeFalse$
+          pkg: ^github.com/onsi/gomega$
+          msg: "it does not produce a good failure message - use BeFalseBecause with an explicit printf-style failure message instead, or plain Go: if ... { ginkgo.Fail(...) }"        
       {{- end}}
-  {{- if .Base }}
-  stylecheck:
-    checks:
-      - "ST1019"   # Importing the same package multiple times
-  {{- end}}
-  testifylint:
-    enable-all: true
-    {{- if not .Hints }}
-    disable:  # TODO: remove each disabled rule and fix it
       {{- if .Base }}
-      - float-compare
-      {{- end}}
-      - require-error
+    gocritic:	
+      enabled-checks:	
+        - equalFold	
+        - boolExprSimplify	
     {{- end}}
+    revive:
+      # Only these rules are enabled.
+      rules:
+        - name: exported
+          arguments:
+            - disableStutteringCheck
+    staticcheck:
+      checks:
+        - "all"
+        {{- if .Base }}
+        - "-QF1001"  # TODO(fix) Apply De Morgan’s law
```

Comment:
I'm assuming all of these disable entries are needed because there is code not passing the check?

I don't think we really plan to fix any of these findings in the existing code. Let's remove `TODO(fix) ` and leave just the short explanation what each entry is about?